File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        frequency = {}
        largest_freq = 0
        for num in nums:
            frequency[num] = frequency.get(num, 0) + 1
            largest_freq = max(largest_freq, frequency[num])

        bucket = [[] for i in range(largest_freq + 1)]

        for key, val in frequency.items():
            bucket[val].append(key)

        output = []
        for j in reversed(bucket):
            for m in j:
                output.append(m)
                if len(output) == k:
                    return output

        return None

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const numsSet = new Set(nums);
        let output = 0;

        for (let num of numsSet) {
            if (!numsSet.has(num - 1)) {
                let length = 1;
                while (numsSet.has(++num)) {
                    length++;
                }

                output = Math.max(output, length);
            }
        }
        return output;
    }
}
